### PR TITLE
Document missing deny by default lints

### PR DESCRIPTION
**Pull request by 197g**

### CODE REVIEW

File: src/doc/rustc/src/lints/listing/deny-by-default.md [hunk 1]
```diff
@@ -222,3 +222,33 @@ error: invalid `crate_type` value
...
+  |         ^^^^^^^^^
+  |
+```
+
+## order-dependent-trait-objects
+
+This lint detects a trait coherency violation that would allow creating two
+trait impls for the same dynamic trait object involving marker traits.
+
+## duplicate-macro-exports
```
Comment (Centril) [inline src/doc/rustc/src/lints/listing/deny-by-default.md]: This is getting nixed soon so no need to add it here.

Comment (HeroicKatora) [reply src/doc/rustc/src/lints/listing/deny-by-default.md]: Removed the `duplicate-macro-exports` section

**Comment by jhpratt** [comment]
Ideally, _all_ of the documentation for lints should be updated. There are a number that are out of date or outright missing (like `rust_2018_idioms`, which encompasses a few others). At least for me, it's somewhat of a guessing game to figure out which one belongs where.

**Comment by JohnCSimon** [comment]
Ping from triage
@GuillaumeGomez can you please review this pr?
cc: @HeroicKatora 

thanks.

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**